src/azure-cli/azure/cli/command_modules/serviceconnector/_utils.py

```diff
@@ -53,46 +61,52 @@ def generate_random_string(length=5, prefix='', lower_only=False, ensure_complex
         character_set = string.ascii_lowercase
 
     while True:
-        randstr = '{}{}'.format(prefix, ''.join(random.sample(character_set, length)))
+        randstr = '{}{}'.format(prefix, ''.join(
+            random.sample(character_set, length)))
         lowers = [c for c in randstr if c.islower()]
         uppers = [c for c in randstr if c.isupper()]
         numbers = [c for c in randstr if c.isnumeric()]
         if not ensure_complexity or (lowers and uppers and numbers):
             break
 
-    return randstr
+    return randstr.lower()
```

Comment:
why making it lower forcely
this changed the behavior of the function, and possibly break the existing callers of the function